include/deal.II/lac/petsc_precondition.h

```diff
@@ -146,6 +146,14 @@ namespace PETScWrappers
                         const AdditionalData &additional_data = AdditionalData());
 
     /**
+     * Same as above but without setting a matrix to form the preconditioner.
+     * Intended to be used with SLEPc objects.
+     */
+    PreconditionJacobi (const MPI_Comm communicator,
+                        const AdditionalData &additional_data = AdditionalData());
```

Comment:
i tried step-40 with the following patch
```
diff --git a/step-40.cc b/step-40.cc
index 52b02b2..9334a07 100644
--- a/step-40.cc
+++ b/step-40.cc
@@ -478,8 +478,6 @@ namespace Step40
     LA::SolverCG solver(solver_control);
 #endif

-    LA::MPI::PreconditionAMG preconditioner;
-
     LA::MPI::PreconditionAMG::AdditionalData data;

 #ifdef USE_PETSC_LA
@@ -487,7 +485,7 @@ namespace Step40
 #else
     /* Trilinos defaults are good */
 #endif
-    preconditioner.initialize(system_matrix, data);
+    LA::MPI::PreconditionAMG preconditioner(mpi_communicator,data);

     solver.solve (system_matrix, completely_distributed_solution, system_rhs,
                   preconditioner);
```
and the results are not pleasant:
```
Cycle 0:
   Number of active cells:       1024
   Number of degrees of freedom: 4225
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: Arguments are incompatible
[0]PETSC ERROR: Both n and N cannot be PETSC_DECIDE
  likely a call to VecSetSizes() or MatSetSizes() is wrong.
See http://www.mcs.anl.gov/petsc/documentation/faq.html#split
```
So one should not use those methods in solvers and error is cryptic...